packages/core/base/src/signer.ts

```diff
@@ -1,16 +1,56 @@
-import type { Connection, Transaction, TransactionSignature } from '@solana/web3.js';
+import { Connection, SendOptions, Transaction, TransactionSignature, VersionedTransaction } from '@solana/web3.js';
 import type { SendTransactionOptions, WalletAdapter } from './adapter.js';
 import { BaseWalletAdapter } from './adapter.js';
 import { WalletSendTransactionError, WalletSignTransactionError } from './errors.js';
 
 export interface SignerWalletAdapterProps {
     signTransaction(transaction: Transaction): Promise<Transaction>;
+    signVersionedTransaction(transaction: VersionedTransaction): Promise<VersionedTransaction>;
     signAllTransactions(transaction: Transaction[]): Promise<Transaction[]>;
+    signAllVersionedTransactions(transactions: VersionedTransaction[]): Promise<VersionedTransaction[]>;
 }
 
 export type SignerWalletAdapter = WalletAdapter & SignerWalletAdapterProps;
 
 export abstract class BaseSignerWalletAdapter extends BaseWalletAdapter implements SignerWalletAdapter {
+    async sendVersionedTransaction(
+        transaction: VersionedTransaction,
+        connection: Connection,
+        options: SendOptions = {}
+    ): Promise<TransactionSignature> {
+        let emit = true;
+        try {
+            const version = transaction.message.version;
+            if (this.supportedTransactionVersions === null) {
+                throw new WalletSendTransactionError(`Sending versioned transactions isn't supported by this wallet`);
+            } else if (!this.supportedTransactionVersions.has(version)) {
+                throw new WalletSendTransactionError(
+                    `Sending transaction version ${version} isn't supported by this wallet`
+                );
+            }
```

Comment:
I'm wary of increasing the public API of Wallet Adapter so much, especially with the Wallet Standard work being the primary focus now. Based on how this is implemented (supported versions exposed and runtime checks for support rather than complete type safety), I think we could just change `sendTransaction`, `signTransaction`, and `signAllTransactions` to accept `Transaction | VersionedTransaction` instead of having new public methods for all of them. Maybe we can keep these methods protected, and have the public methods call them if they are called with a `VersionedTransaction`.

Reply:
I've consolidated `VersionedTransaction` support into `sendTransaction` as you suggested but `signTransaction` has trickier typing because it returns the transaction

Reply:
Ah, yeah. I think we can do this with

```ts
<T extends Transaction | VersionedTransaction>signTransaction(transaction: T, ...): Promise<T>;
<T extends Transaction | VersionedTransaction>signAllTransactions(transactions: T[], ...): Promise<T[]>;
```

though this would also be conditional if https://github.com/solana-labs/wallet-adapter/pull/558#discussion_r969010739 is implemented.